dask_sql/physical/rel/logical/subquery_alias.py

```diff
@@ -16,4 +17,20 @@ class SubqueryAlias(BaseRelPlugin):
 
     def convert(self, rel: "LogicalPlan", context: "dask_sql.Context"):
         (dc,) = self.assert_inputs(rel, 1, context)
-        return dc
+
+        cc = dc.column_container
+
+        alias = rel.subquery_alias().getAlias()
+
+        return DataContainer(
+            dc.df,
+            cc.rename(
+                {
+                    col: renamed_col
+                    for col, renamed_col in zip(
+                        cc.columns,
+                        (f"{alias}.{col.split('.')[-1]}" for col in cc.columns),
+                    )
+                }
+            ),
+        )
```

Comment:
With this DataFusion bump, we now optimize otherwise identical projections with different aliases into a single projection followed by the equivalent `SubqueryAlias`, e.g. turning

```
Projection: customer_total_return.ctr_store_sk, customer_total_return.ctr_total_return, alias=ctr2
  Projection: store_returns.sr_store_sk AS ctr_store_sk, SUM(store_returns.sr_fee) AS ctr_total_return, alias=customer_total_return
```

into

```
SubqueryAlias: ctr2
  SubqueryAlias: customer_total_return
    Projection: store_returns.sr_store_sk AS ctr_store_sk, SUM(store_returns.sr_fee) AS ctr_total_return
```

This exposed the fact that our `SubqueryAlias` plugin was a no-op that didn't actually do the required aliasing. These modifications should unblock failures stemming from this optimization.